lib/experimental/interactivity-api/blocks.php

```diff
@@ -230,15 +230,15 @@ function gutenberg_block_core_navigation_add_directives_to_submenu( $w ) {
 add_filter( 'render_block_core/navigation', 'gutenberg_block_core_navigation_add_directives_to_markup', 10, 1 );
 
 /**
- * Replaces view script for the File and Navigation blocks with version using Interactivity API.
+ * Replaces view script for the File, Navigation, and Image blocks with version using Interactivity API.
  *
  * @param array $metadata Block metadata as read in via block.json.
  *
  * @return array Filtered block type metadata.
  */
 function gutenberg_block_update_interactive_view_script( $metadata ) {
 	if (
-		in_array( $metadata['name'], array( 'core/file', 'core/navigation' ), true ) &&
+		in_array( $metadata['name'], array( 'core/file', 'core/navigation', 'core/image' ), true ) &&
```

Comment:
We're adding the interactivity scripts for the image block here.

Reply:
I am not familiar with the scripts enqueuing. Maybe @gziolo can help here better. The two solutions that come to my mind are:

* Remove the `core/image` here and enqueue it conditionally in the `index.php` of the image. When [this condition](https://github.com/artemiomorales/gutenberg/blob/experiment/image-with-interactivity-api/packages/block-library/src/image/index.php#L50) is true.
* Dequeue the script using something like [wp_dequeue_script](https://developer.wordpress.org/reference/functions/wp_dequeue_script/) when that condition is false.

Reply:
The plan is to optimize it later. You need to do it globally because one Image block could use the lightbox, and another might not. That's why you can't decide on a single-block level. As of today, when `viewScript` is defined for the block in `block.json` then all listed files are always enqueued, and that's what everyone expects.